# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/events/PointerEventHandler.h
# Commit: 5718f7f3067e
# Full Hash: 5718f7f3067eff42e097a6013acd82fbe6aa299a
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2025-05-15 22:58:00
# Regressor Bug: 1550462
# File Overlap Count: 1
# Description:
#   Bug 1550462 - part 2: Make `PresShell::HandleEvent` dispatch preceding `pointerrawupdate` event r=smaug,dom-core,edgar
#   
#   This patch tries to dispatch `ePointerRawUpdate` with
#   `PresShell::EventHandler::DispatchPrecedingPointerEvent` as same as usual
#   pointer events.
# ==============================================================================

diff -r aae164cff6aa -r 5718f7f3067e dom/events/PointerEventHandler.h
--- a/dom/events/PointerEventHandler.h	Thu May 15 03:20:28 2025 +0000
+++ b/dom/events/PointerEventHandler.h	Thu May 15 03:20:29 2025 +0000
@@ -270,14 +270,16 @@
       const WidgetPointerEvent& aSourceEvent);
 
   static bool ShouldGeneratePointerEventFromMouse(WidgetGUIEvent* aEvent) {
-    return aEvent->mMessage == eMouseDown || aEvent->mMessage == eMouseUp ||
+    return aEvent->mMessage == eMouseRawUpdate ||
+           aEvent->mMessage == eMouseDown || aEvent->mMessage == eMouseUp ||
            (aEvent->mMessage == eMouseMove &&
             aEvent->AsMouseEvent()->IsReal()) ||
            aEvent->mMessage == eMouseExitFromWidget;
   }
 
   static bool ShouldGeneratePointerEventFromTouch(WidgetGUIEvent* aEvent) {
-    return aEvent->mMessage == eTouchStart || aEvent->mMessage == eTouchMove ||
+    return aEvent->mMessage == eTouchRawUpdate ||
+           aEvent->mMessage == eTouchStart || aEvent->mMessage == eTouchMove ||
            aEvent->mMessage == eTouchEnd || aEvent->mMessage == eTouchCancel ||
            aEvent->mMessage == eTouchPointerCancel;
   }
@@ -290,11 +292,18 @@
 
   static bool IsDragAndDropEnabled(WidgetMouseEvent& aEvent);
 
- private:
   // Get proper pointer event message for a mouse or touch event.
-  static EventMessage ToPointerEventMessage(
+  [[nodiscard]] static EventMessage ToPointerEventMessage(
       const WidgetGUIEvent* aMouseOrTouchEvent);
 
+  /**
+   * Return true if the window containing aDocument has had a
+   * `pointerrawupdate` event listener.
+   */
+  [[nodiscard]] static bool NeedToDispatchPointerRawUpdate(
+      const dom::Document* aDocument);
+
+ private:
   // Set pointer capture of the specified pointer by the element.
   static void SetPointerCaptureById(uint32_t aPointerId,
                                     dom::Element* aElement);